lib/src/bindings.dart

```diff
@@ -217,3 +225,36 @@ abstract class DeltaDocumentSnapshot implements admin.DocumentSnapshot {
   /// Gets the value for a given key.
   external dynamic get(key);
 }
+
+@JS()
+@anonymous
+abstract class PubsubFunctions {
+  /// Registers a function that triggers on Pubsub write events to
+  /// the [topic].
+  external TopicBuilder topic(String topic);
+}
+
+/// The Pubsub topic builder interface.
+@JS()
+@anonymous
+abstract class TopicBuilder {
+  /// Event handler that fires every time an event is public in Pubsub.
+  external CloudFunction onPublish(void handler(Event<Message> event));
+}
+
+/// Interface representing a Google Cloud Pub/Sub message.
+@JS()
+@anonymous
+abstract class Message {
+  /// User-defined attributes published with the message, if any.
+  external Map<String, String> get attributes;
```

Comment:
Not sure if I'm looking at old diff but I think declaring attributes as a Map on this JS interop class is a bit misleading and may break in the future (if Dart JS compilers start type checking this). Strictly speaking `jsObject is! Map`, so I'd still make this `dynamic`.